File: example/ios/Runner/Biz/PowerImageAssetsImageLoader.m
//
//  MyAssetsImageLoader.m
//  Runner
//
//  Created by 王振辉 on 2021/7/20.
//

#import "PowerImageAssetsImageLoader.h"
#import <UIKit/UIKit.h>

@implementation PowerImageAssetsImageLoader

- (void)handleRequest:(PowerImageRequestConfig *)requestConfig completed:(PowerImageLoaderCompletionBlock)completedBlock {
    UIImage *image = [UIImage imageNamed:requestConfig.srcString];
    if (image) {
        completedBlock([PowerImageResult successWithImage:image]);
    }else {
        completedBlock([PowerImageResult failWithMessage:@"MyAssetsImageLoader UIImage imageNamed: nil"]);
    }
}

@end

File: example_kotlin_swift/ios/Runner/Loaders/PowerImageAssetsImageLoader.swift
//
//  PowerImageAssetsImageLoader.swift
//  Runner
//
//  Created by wjr on 2022/5/30.
//

import power_image

class PowerImageAssetsImageLoader: NSObject, PowerImageLoaderProtocol{
    
    func handleRequest(_ requestConfig: PowerImageRequestConfig!, completed completedBlock: PowerImageLoaderCompletionBlock!) {
        
        let image = UIImage(named: requestConfig.srcString())
        
        if let image = image {
            completedBlock(PowerImageResult.success(with: image))
        }else{
            completedBlock(PowerImageResult.fail(withMessage: "PowerImageAssetsImageLoaderError!"))
        }
    }
}
